src/frontend/js/index.tsx

```diff
@@ -47,11 +47,49 @@ function isComponentName(
 }
 
 // Wait for the DOM to load before we scour it for an element that requires React to be rendered
-document.addEventListener('DOMContentLoaded', event => {
+document.addEventListener('DOMContentLoaded', async event => {
   // Find all the elements that need React to render a component
-  Array.prototype.forEach.call(
+  const richieReactSpots = Array.prototype.slice.call(
     document.querySelectorAll('.richie-react'),
-    async (element: Element) => {
+  );
+
+  // Only move on with anything if there are actually components to render
+  if (richieReactSpots.length) {
+    // Determine the BCP47/RFC5646 locale to use, defaulting to en-US
+    const locale =
+      document.querySelector('html')!.getAttribute('lang') || 'en-US';
```

Comment:
the || 'en-US' defeats our purpose as lang is required to be set?

Reply:
You're right, it should probably just break in this case